cachito/workers/tasks/general.py

```diff
@@ -173,4 +173,35 @@ def filter_git_dir(tar_info):
         # Add the dependencies to the bundle
         bundle_archive.add(str(bundle_dir.deps_dir), "deps")
 
+
+def aggregate_packages_data(request_id: int) -> None:
+    """Aggregate packages data generated for each package manager into one unified data file.
+
+    :param int request_id: the request id.
+    """
+    set_request_state(request_id, "in_progress", "Aggregating packages data")
+    bundle_dir: RequestBundleDir = RequestBundleDir(request_id)
+
+    def _get_packages_data():
+        request = get_request(request_id)
+        for pkg_manager in request["pkg_managers"]:
+            data = PackagesData()
+            data.load(getattr(bundle_dir, f"{pkg_manager}_packages_data"))
+            yield data
+
+    aggregated_data = PackagesData()
+
+    for data in _get_packages_data():
+        aggregated_data += data
+
+    log.debug("Write request %s packages data into %s", request_id, bundle_dir.packages_data)
+    aggregated_data.write_to_file(str(bundle_dir.packages_data))
+
+
+@app.task(priority=10)
+@runs_if_request_in_progress
+def finalize_request_creation(request_id):
+    """Execute tasks to finalize the request creation."""
+    create_bundle_archive(request_id)
+    aggregate_packages_data(request_id)
```

Comment:
Since both of these functions will call `get_request()`, could we instead call that here and pass the entire request dict instead of the `request_id`? The request JSON can be very large in some cases, saving one http request could be significant